### PR TITLE
Makes wound treatment with medical stack items less jank

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Erratas the "assessing injury" do after on medical stack items
- Makes wound treatment just a natural part of medical stack items, meaning you don't do an entirely separate interaction to suture a bleeding wound, and then an entirely separate interaction to treat bruises
- Bleeding wounds will go away immediately upon losing all blood flow, rather than waiting next life tick
- Cauterizing wounds now plays the cautery sfx

## Why It's Good For The Game

- I found the assessing do after really annoying, like damn I already assessed it with my eyes
- It's suuuuper unintuitive that treating bruises is different to treating cuts, there are some instances in which you stop treating the wounds because it downgraded and you just don't notice, it's annoying. This just makes it nice and uniform
- More reactive = easier to work with
- Always found it to be weird

TLDR: Have you ever been suturing someone but you randomly get forced to stop because the wound got better, or have you ever been suturing someone and they started bleeding so you have to stop suturing then start again to actually count as fixing the bleed rather than treating bruises? Yeah that changes that, it's all one do after

## Changelog

:cl: Melbert
qol: Cauterizing bleeding wounds now plays the cautery sfx.
qol: Bleeding wounds will now go away the moment they're fully healed, rather than a second or two later.
qol: Suture / Mesh treatment is now uniform! meaning healing bruises with a suture is now the same thing as healing cuts with a suture. This has very little difference in practice, but it should generally result in a lot smoother experience. 
qol: Suture / Mesh usage has been reworked slightly, and now offers two modes of use: 
qol: LEFT CLICKING will heal in AUTO MODE, which will AUTOMATICALLY switch between damaged bodyparts, prioritizing your targeted limb. You cannot change target mid-heal; changing target simply changes your priority for your NEXT heal.
qol: RIGHT CLICKING will heal in MANUAL MODE, which functions like it does currently - allowing you to change your target before you finish your heal and giving you a 1 second "assessment" step to change your target when you're done healing a limb. Manual mode is 10% faster than Auto mode.
/:cl:

### CODE REVIEW

**Comment by Jezzared** [comment]
Assess injury is supposed to allow you the half a second necessary to switch limb targeting if you need, in order to heal multiple body parts of damage without needing to click the target every time.

**Comment by MrMelbert** [comment]
What if it just automatically swapped to treating another bodypart

**Comment by MrMelbert** [comment]
I made it so it will automatically cycle between all wounded bodyparts when your first targeted bodypart is fully healed, but now I'm wondering if I should like, instead make that an RMB interaction

I can also make it so it *only* pops up the "assessing injury" thing if you finish tending a wound, maybe I make LMB do that and RMB loop? Would like feedback

**Comment by epochayur** [comment]
Great PR, although I would honestly prefer automatically moving to the next body part not be the default behavior, imo takes some of the fun out of manually healing someone like this. 

I think a lot of the fun of the game is job knowledge and mastery, and part of that mastery for medbay is seeing how quickly you can get someone healed up, and I find quickly switching between limbs manually (especially using the numpad) to be very satisfying. I would be sad to see that taken away

**Comment by MatrixOne500** [comment]
Great PR, thank you!

**Comment by MrMelbert** [comment]
> Great PR, although I would honestly prefer automatically moving to the next body part not be the default behavior, imo takes some of the fun out of manually healing someone like this.
> 
> I think a lot of the fun of the game is job knowledge and mastery, and part of that mastery for medbay is seeing how quickly you can get someone healed up, and I find quickly switching between limbs manually (especially using the numpad) to be very satisfying. I would be sad to see that taken away

I don't disagree which is why I'm trying to get as much feedback as possible. I might make "auto loop" on LMB and "manual heal" on RMB, maybe having manual be slightly faster?

**Comment by epochayur** [comment]
> I don't disagree which is why I'm trying to get as much feedback as possible. I might make "auto loop" on LMB and "manual heal" on RMB, maybe having manual be slightly faster?
I think that sounds like a fair compromise

**Comment by MrMelbert** [comment]
I have changed it thusly:

- LMB on a limb will use `auto mode`:
   - Changing targeted limb mid do_after will not affect the current target: it is locked in once you start
   - Fully healing a limb will **automatically** start healing another limb, prioritizing your current target if possible. If your current target can't be healed, it'll just pick another limb, if no limbs can be healed, it stops.
      - This means you can change your target around while healing to decide what to heal *next* rather than what to heal *now*
- RMB on a limb will use `manual mode`:
   - 10% faster than auto mode
   - Changing targeted limb mid do_after will update the current target, but will NOT interrupt the do_after if you change to an invalid limb (meaning it will cancel afterwards, as current behavior). 
      - This means you can change your target on demand
   - Fully healing a limb will do a 1 second "assessing" do_after allowing you to change target. If no new target is picked, you stop healing (as current behavior).

**Comment by Loafin34** [comment]
you should add the auto mode and manual mode to your description and title
